File: languages/go/test-one.go
// The test-one-go tool parses the provided HTML file and print the microformats found.
package main

import (
	"encoding/json"
	"strings"
	"fmt"
	"net/url"
	"os"

	"willnorris.com/go/microformats"
)

func main() {
	baseStr := "http://example.com/"
	if strings.HasPrefix(os.Args[1], "vendor/mf2/tests/tests/microformats-v2-unit/") {
		// This is a unit test; these use a different base URL
		baseStr = "http://example.test"
	}
	baseURL, _ := url.Parse(baseStr)

	f, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
	defer f.Close()

	data := microformats.Parse(f, baseURL)

	json, _ := json.MarshalIndent(data, "", "  ")
	fmt.Println(string(json))
}

File: languages/php/test-one.php
<?php
error_reporting(E_ALL & ~E_DEPRECATED);
if(count($argv) < 2){
    echo "Usage: ". $argv[0]." <inputfile>\n";
    die();
}

require 'vendor/autoload.php';

$file = $argv[1];
$data = file_get_contents($file);
$base = 'http://example.com/';
if (strpos($file, "/microformats-v2-unit/") !== false) {
    // This is a unit test; these use a different base URL
    $base = 'http://example.test';
}

// Use "JSON mode" in the parser, which makes a distinction
//   between empty arrays and empty objects
$p = new Mf2\Parser($data, $base, true);
$output = $p->parse();

echo json_encode($output) . "\n";
